its_on/admin/views/switches.py

```diff
@@ -140,6 +148,45 @@ async def post(self) -> Dict[str, Dict]:
         raise web.HTTPFound(location=location)
 
 
+class SwitchesCopyAdminView(web.View, CreateMixin):
+    validator = SwitchAddFromAnotherItsOnAdminPostRequestSchema()
+    model = switches
+
+    @staticmethod
+    async def _get_switches_data() -> MultiDictProxy:
+        async with ClientSession() as session:
+            async with session.get(settings.SYNC_FROM_ITS_ON_URL) as resp:
+                return await resp.json()
+
+    @login_required
+    async def post(self) -> None:
```

Comment:
I don't really like it when some kind of business logic is written directly in the body of the view. How about moving the logic to some kind of container?

Reply:
This code uses methods of the class and it will be hard to move this logic out of the class without refactoring of base structures.